Orange/data/table.py

```diff
@@ -1625,10 +1626,16 @@ def transpose(cls, table, feature_names_column="", meta_attr_name="Feature name"
         # - classes and metas to attributes of attributes
         # - arbitrary meta column to feature names
         self.X = table.X.T
-        attributes = [ContinuousVariable(str(row[feature_names_column]))
-                      for row in table] if feature_names_column else \
-            [ContinuousVariable(feature_name + " " + str(i + 1).zfill(
-                int(np.ceil(np.log10(n_cols))))) for i in range(n_cols)]
+        if feature_names_column:
+            names, attributes = [], []
+            for row in table:
+                name = get_unique_names(names, str(row[feature_names_column]))
```

Comment:
This has quadratic complexity -- `get_unique_names` is linear. My suggestion to use `get_unique_names` was probably bad. It could be better to use a dict (or `Counter`) to count the occurrences of each name.